### DISCUSSION TITLE
Help writing a new lint test

### DISCUSSION THREAD

**Discussion by matthewsht** [warpnet/salt-lint#289] [Q&A] [answered]
My team has a team specific formating for state names:

$FORUMLA - $STATE - $OPT_SUBSTATE - description

Based on the filename, I think I can figure out what the formula, state and option substate names should be.

My question is - how do I identify the state name in the text? Is that even regexable?

Suggestions welcome.

**Comment by roaldnefs** [comment]
I've converted the GitHub issue (#281) to a discussion.

**Comment by roaldnefs** [comment, marked as answer]
If you filter out all the Jinja placeholders and comments you could easily identify the state names in a SLS file using the regex `^(\S[^:]+):`. This will get you all the lines starting without a whitespace (indentation) up to the `:` character in a capture group.

Using this regex you could write a rule that checks if the found state name starts with the formula, state and option substate you already know based upon the filename. If one of the state names doesn't match, it's not following your teams convention and should return the found regex capture group(s).

The filename can be obtain the filename inside your `match(self, file, line)` function using the `file['path']` variable.
